vulnerabilities/importers/istio.py

```diff
@@ -0,0 +1,173 @@
+            if processed_data:
+                advisories.extend(processed_data)
+        return self.batch_advisories(advisories)
+
+    def added_advisories(self) -> Set[Advisory]:
+        files = self._added_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.extend(processed_data)
+        return self.batch_advisories(advisories)
+
+    def get_versions_for_pkg_from_range_list(self, version_range_list):
+        # Takes a list of version ranges(affected) of a package
+        # as parameter and returns a tuple of safe package versions and
+        # vulnerable package versions
+
+        safe_pkg_versions = []
+        vuln_pkg_versions = []
+        all_version_list = self.version_api.get("istio/istio")
+        if not version_range_list:
+            return all_version_list, []
+        version_ranges = {RangeSpecifier(r) for r in version_range_list}
+        for version in all_version_list:
+            if any([version in v for v in version_ranges]):
+                vuln_pkg_versions.append(version)
+
+        safe_pkg_versions = set(all_version_list) - set(vuln_pkg_versions)
+        return safe_pkg_versions, vuln_pkg_versions
+
+    def get_data_from_md(self, file):
+        data = {}
+        for line in file:
+            line = line.strip()
+            line = line.split()
+            if len(line) > 0 and line is not None:
+
+                start = line[0]
+
+                if start == "title:":
+                    data["title"] = " ".join(line[1:])
+                elif start == "description:":
+                    data["description"] = " ".join(line[1:])
+                elif start == "cves:":
+                    data["cves"] = " ".join(line[1:])
+                    data["cves"] = data["cves"].replace("[", "")
+                    data["cves"] = data["cves"].replace("]", "")
+                    data["cves"] = data["cves"].split(",")
+
+                elif start == "releases:":
+                    data["releases"] = " ".join(line[1:])
+                    data["releases"] = data["releases"].replace("[", "")
+                    data["releases"] = data["releases"].replace("]", "")
+                    data["releases"] = data["releases"].replace('"', "")
+                    data["releases"] = data["releases"].split(",")
+        releases = []
+        if data.get("releases"):
+            for release in data["releases"]:
+                release = release.strip()
+                release = release.split(" ")
+                if len(release) > 2:
+                    lbound = ">=" + release[0]
+                    ubound = "<=" + release[2]
+                    releases.append(lbound + "," + ubound)
+        data["releases"] = releases
+
+        return data
+
+    def process_file(self, path):
+
+        advisories = []
+
+        with open(path) as f:
+            data = {}
```

Comment:
Why declare this, when it is being are captured  below. This makes it redundant

Reply:
Ok